### PR TITLE
feat: refactor nav and add admin menu

### PR DESCRIPTION
**Pull request by wpf500**
Implementation of the admin nav. I have used this as an opportunity to refactor the menu code:

* Remove magic 240px by adding it as a `menu` spacing option to Tailwind
* Reworked how styles are applied so there is no custom CSS
* Separated sections out into into separate components

image

### CODE REVIEW

**Review by rlafuente — APPROVED** [review]
This is great!

File: src/layouts/menu/TheMenuList.vue [hunk 1]
```diff
@@ -42,56 +38,20 @@
...
+import { canAdmin } from '../../utils/currentUserCan';
 import { logout } from '../../utils/api/auth';
 
-const { t } = useI18n();
+import { menu, adminMenu } from './menu-list';
 
-defineProps<{
-  sections: MenuSection[];
-}>();
+adminMenu;
```
Comment (basilesimon) [inline src/layouts/menu/TheMenuList.vue]: Sorry, I don't know why this is needed?

**Review by basilesimon — CHANGES_REQUESTED** [review]

File: src/layouts/menu/TheMenuListItem.vue [hunk 2]
```diff
@@ -1,16 +1,46 @@
...
+          'text-primary-80 hover:bg-primary-5',
+          'text-primary-80 bg-primary-20',
+        ],
+        settings: [
+          'text-grey-dark hover:bg-grey-lighter',
+          'bg-primary-5 text-grey-darker',
+        ],
+      } as const
+    )[props.type][props.isActive ? 1 : 0])
+);
```
Comment (basilesimon) [inline src/layouts/menu/TheMenuListItem.vue]: 🥲

File: src/layouts/menu/menu-list.ts [hunk 3]
```diff
@@ -44,4 +40,31 @@ export default [
...
+      },
+      {
+        title: t('menu.notices'),
+        href: '/admin/notices',
+        icon: ['fa', 'bullhorn'],
+        isActive: /^\/admin\/notices.*/,
+      },
+    ],
+  },
+];
```
Comment (basilesimon) [inline src/layouts/menu/menu-list.ts:70]: Also, and kind of related to another thread on the admin dashboard:

I would really see the dashboard being at the top of the list, and potentially either the default page or at the root of the route (geddit?)

File: src/layouts/menu/TheMenuList.vue (hunk 1 above)
Comment (wpf500) [reply src/layouts/menu/TheMenuList.vue]: Oops will remove, thanks!

File: src/layouts/menu/menu-list.ts (hunk 3 above)
Comment (wpf500) [reply src/layouts/menu/menu-list.ts:70]: Yeah I agree on all points. The "Dashboard" I have in the menu at the moment is actually just a link to the legacy app (as some functionality still only exists there). It needs a better name so we can add the actual dashboard to the menu!

File: src/layouts/menu/TheMenuListItem.vue (hunk 2 above)
Comment (wpf500) [reply src/layouts/menu/TheMenuListItem.vue]: This is very strange code, could it be clearer somehow? Maybe defining the object outside the `computed`?

```ts
const itemClasses = {
  main: [
    'text-primary-80 hover:bg-primary-5',
    'text-primary-80 bg-primary-20',
  ],
  settings: [
    'text-grey-dark hover:bg-grey-lighter',
    'bg-primary-5 text-grey-darker',
  ],
} as const;

const blah = computed(() => itemClasses[props.type][props.isActive ? 1 : 0]);
```

Comment (basilesimon) [reply src/layouts/menu/TheMenuListItem.vue]: 👍

**Review by basilesimon — APPROVED** [review]

**Pull request merged**